Samples/PowerShell/functions/func_ModernDashboards.ps1

```diff
@@ -4,6 +4,134 @@
+
+    The first command will (most likely) return nothing as DashboardId 1 is a "system" dashboard
+    The second command will return the details of Dashboard ID 1
+
+.NOTES
+    Author:  Kevin M. Sparenberg (https://thwack.solarwinds.com/members/kmsigma)
+    Version: 0.9
+    Last Updated: 2021-09-30
+    Validated: Orion Platform 2020.2.5
+
+    TBD List:
+        * Validate that the script works on non-Windows
+#>
+function Get-SwisModernDashboard {
+    [CmdletBinding(
+        DefaultParameterSetName = 'Normal', 
+        PositionalBinding = $false,
+        HelpUri = 'https://documentation.solarwinds.com/en/success_center/orionplatform/content/core-fusion-dashboard-import-export.htm',
+        ConfirmImpact = 'Low')]
+    [Alias('Get-ModernDashboard')]
+    [OutputType()]
+    Param
+    (
+        # The connection to the SolarWinds Information Service
+        [Parameter(
+            Mandatory = $false, 
+            ValueFromPipeline = $true,
+            ValueFromPipelineByPropertyName = $true, 
+            ValueFromRemainingArguments = $false)]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
+        [Alias("Swis")] 
+        [SolarWinds.InformationService.Contract2.InfoServiceProxy]$SwisConnection = $Global:SwisConnection,
+
+        # The dashboard Id we'll display
+        [AllowNull()]
+        [Alias("Id")] 
+        [int32[]]$DashboardId,
+
+        # Filter for specific owners?
+        [AllowNull()]
+        [string[]]$Owner,
+
+        # Should we include system Dashboards?
+        [AllowNull()]
+        [switch]$IncludeSystem,
+
+        # Omits white space and indented formatting in the output string.
+        [AllowNull()]
+        [switch]$IncludeJson
+
+    )
+
+    Begin {
+        # Define the Property List
+        $PropertyList = "DashboardID", "DisplayName", "UniqueKey", "Owner", "LastUpdate", "Private"
+
+        # Define default Query
+        $Swql = "SELECT DashboardID, ParentID, Owner, UniqueKey, LastUpdate, Private, IsSystem, DisplayName FROM Orion.Dashboards.Instances WHERE ParentID IS NULL"
+
+    }
+    Process {
+        
+        #region Filter for DashboardIds
+        # Add Dashboards if we are sent IDs
+        if ( $DashboardId ) {
+            # Add an = to the WHERE clause
+            $Swql += " AND DashboardID = $( $DashboardId )"
+        }
+        #endregion Filter for DashboardIds
+        
+        #region Filter for Owners
+        # If the Owner parameter is passed, we need to filter for that as well
+        # Same logic as above, but for the array, we need to be sure to include the single quotes as wrappers
+        if ( $Owner ) {
+            $Swql += " AND Owner = '$( $DashboardId )'"
+        }
+        #endregion Filter for Owners
+
+        # If we want to skip System Views, then we filter for 'IsSystem' is False
+        if ( -not $IncludeSystem ) {
+            $Swql += " AND IsSystem = 'FALSE'"
+            $PropertyList += "IsSystem"
```

Comment:
For consistency, it might be best to always have `IsSystem` in the `$PropertyList`.  But if not, wouldn't it make more sense to include it when `$IncludeSystem` is true, where you could have a mix of system and user dashboards in the results?  It seems to me like you'd want to move this line to an `else` block, unless I'm missing something.

Reply:
Yep - logic fail on my part.  Forgot that I was doing the opposite check to other parameters.